backend/danswer/connectors/gitlab/connector.py

```diff
@@ -0,0 +1,174 @@
+
+import itertools
+from collections.abc import Iterator
+from datetime import datetime
+from datetime import timezone
+from typing import Any
+
+import gitlab
+from danswer.configs.app_configs import INDEX_BATCH_SIZE
+from danswer.configs.constants import DocumentSource
+from danswer.connectors.interfaces import GenerateDocumentsOutput
+from danswer.connectors.interfaces import LoadConnector
+from danswer.connectors.interfaces import PollConnector
+from danswer.connectors.interfaces import SecondsSinceUnixEpoch
+from danswer.connectors.models import ConnectorMissingCredentialError
+from danswer.connectors.models import Document
+from danswer.connectors.models import Section
+from danswer.utils.logger import setup_logger
+
+
+logger = setup_logger()
+
+
+def _batch_gitlab_objects(
+    git_objs: list[Any],
+    batch_size: int
+) -> Iterator[list[Any]]:
+    it = iter(git_objs)
+    while True:
+        batch = list(itertools.islice(it, batch_size))
+        if not batch:
+            break
+        yield batch
+
+def _convert_merge_request_to_document(mr: Any) -> Document:
+    return Document(
+        id=mr.web_url,
+        sections=[Section(link=mr.web_url, text=mr.description or "")],
+        source=DocumentSource.GITLAB,
+        semantic_identifier=mr.title,
+        # updated_at is UTC time but is timezone unaware, explicitly add UTC
+        # as there is logic in indexing to prevent wrong timestamped docs
+        # due to local time discrepancies with UTC
```

Comment:
Could we also add the Author as the primary_owners, and anyone who comments as the secondary_owners? Thanks!

Reply:
Owner is type of BasicExpertInfo, which will need email but we don't get email. is it okay if i add only with name and username as display name ?